### PR TITLE
Fix for CANFAR specific menu items.

### PR DESCRIPTION
**Pull request by at88mph**
Menu item lookup is done by URI now for compatibility with the Registry entries.
State setting is always merged, so no need to get the old state to add to and set it.

### CODE REVIEW

File: src/react/canfar/CanfarLoginModal.js [hunk 1]
```diff
@@ -75,12 +77,12 @@ class CanfarLoginModal extends React.Component {
                          placeholder="Password"/>
                 </div>
                 {errMsg}
-                <a href={this.state.modalURLs.passreset}
+                <a href={this.state.modalURLs[PASS_RESET_URI]}
```
Comment (SharonGoliath) [inline src/react/canfar/CanfarLoginModal.js:80]: Is `passreset` still required in `HEADER_URL_DEFAULTS`, if it's no longer referenced in this location?

I have the same question for `acctrequest`, `groups`, and `search`.

**Review by SharonGoliath — DISMISSED** [review]

File: src/react/canfar/CanfarLoginModal.js (hunk 1 above)
Comment (at88mph) [reply src/react/canfar/CanfarLoginModal.js:80]: Ah, good catch.  I've updated those to match the URIs.

File: src/App.js [hunk 2]
```diff
@@ -49,12 +49,12 @@ const URLS = {
 }
 
 const HEADER_URL_DEFAULTS = {
-  "acctrequest": "",
-  "acctupdate": "",
-  "passreset": "",
-  "passchg": "",
-  "gmui": "",
-  "search": "",
+  "ivo://cadc.nrc.ca/acctrequest": "",
```
Comment (SharonGoliath) [inline src/App.js:52]: There are constants defined for some of these values in `CanfarNavbar.js`. Should `App.js` use those constants?

Comment (at88mph) [reply src/App.js:52]: The `App.js` file uses the `headerURLs` variable to distribute to other React components, so no, it's just storing them to use in building other things.

**Review by SharonGoliath — APPROVED** [review]

**Pull request merged**